torch_xla/csrc/xla_sharding_util.h

```diff
@@ -99,6 +99,14 @@ class ShardingUtil {
                             const xla::OpSharding sharding,
                             const std::vector<std::string>& devices);
 
+  // Returns the indices for the shards. Supports `OTHER` sharding types and
+  // called when input is sharded along the batch axis.
+  static std::vector<std::vector<at::indexing::TensorIndex>>
+  GetShardIndicesForMinibatchTensor(const std::vector<int64_t>& shard_shape,
```

Comment:
shard_shape is the local shard shape for the host. tensor_shape is the global tensor shape. devices is the tpu devices for the local host.